pkg/cache/tas_cache_test.go

```diff
@@ -1318,10 +1353,43 @@ func TestFindTopologyAssignment(t *testing.T) {
 			},
 			enableTASLeastAllocated: true,
 		},
+		"no assignment as node does not have enough allocatable pods (.status.allocatable['pods']); LeastAllocated": {
+			nodes: []corev1.Node{
+				*testingnode.MakeNode("b1-r1-x1").
+					Label("zone", "zone-a").
+					Label(corev1.LabelHostname, "x1").
+					StatusAllocatable(corev1.ResourceList{
+						corev1.ResourceCPU:  resource.MustParse("1000m"),
+						corev1.ResourcePods: resource.MustParse("1"),
+					}).
+					Ready().
+					Obj(),
+			},
+			pods: []corev1.Pod{
+				*testingpod.MakePod("test-running", "test-ns").
+					NodeName("b1-r1-x1").
+					StatusPhase(corev1.PodRunning).
+					Request(corev1.ResourceCPU, "300m").
+					Obj(),
+			},
+			request: kueue.PodSetTopologyRequest{
+				Required: ptr.To(corev1.LabelHostname),
+			},
+			nodeLabels: map[string]string{
+				"zone": "zone-a",
+			},
+			levels: defaultOneLevel,
+			requests: resources.Requests{
+				corev1.ResourceCPU: 300,
+			},
+			count:                   1,
+			wantReason:              `topology "default" doesn't allow to fit any of 1 pod(s)`,
+			enableTASLeastAllocated: true,
+		},
 	}
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
-			ctx := context.Background()
+			ctx, _ := utiltesting.ContextWithLog(t)
```

Comment:
This was added as we can easily debug something powered by logging.

Reply:
Please split out to separate PR.

Reply:
SGTM

Reply:
This was extracted to https://github.com/kubernetes-sigs/kueue/pull/4274